src/cogent3/core/sequence.py

```diff
@@ -144,14 +144,22 @@ def to_json(self):
         """returns a json formatted string"""
         return json.dumps(self.to_rich_dict())
 
-    def translate(self, *args, **kwargs):
+    def translate(self, *args, **kwargs):  # pragma: no cover
         """returns the result of call str.translate
 
         Notes
         -----
         This is a string method, nothing to do with translating into a
         protein sequence.
         """
+        from cogent3.util.warning import discontinued
+
+        discontinued(
+            "function",
+            "cogent3.core.sequence.SequenceI.translate",
+            "06/2023",
```

Comment:
- change `"cogent3.core.sequence.SequenceI.translate"` to `"cogent3.core.sequence.Sequence.translate"` (note the deletion of `I`.
- the version should be `"2023.6"`